indexing-service/src/main/java/io/druid/indexing/common/task/IndexTask.java

```diff
@@ -151,6 +165,8 @@ public IndexTask(
     super(
         // _not_ the version, just something uniqueish
         makeId(id, ingestionSchema),
+        makeId(id, ingestionSchema),
```

Comment:
So I've gone back to the way I initially had it, with the taskResource logic pushed up into AbstractFixedIntervalTask - doing so allows for the passed id value to be reused for groupId _and_ for generating a new taskResource if passed as null.